MoltenVK/MoltenVK/GPUObjects/MVKDeviceMemory.mm

```diff
@@ -412,24 +277,30 @@
 		auto& xmProps = getPhysicalDevice()->getExternalImageProperties(VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLTEXTURE_BIT_KHR);
 		requiresDedicated = requiresDedicated || mvkIsAnyFlagEnabled(xmProps.externalMemoryFeatures, VK_EXTERNAL_MEMORY_FEATURE_DEDICATED_ONLY_BIT);
 	}
-	if (requiresDedicated && !_isDedicated) {
+	if (requiresDedicated && (_dedicatedResourceType == DedicatedResourceType::NONE)) {
 		setConfigurationResult(reportError(VK_ERROR_INITIALIZATION_FAILED, "vkAllocateMemory(): External memory requires a dedicated VkBuffer or VkImage."));
 	}
 }
 
 MVKDeviceMemory::~MVKDeviceMemory() {
-    // Unbind any resources that are using me. Iterate a copy of the collection,
-    // to allow the resource to callback to remove itself from the collection.
-    auto buffCopies = _buffers;
-    for (auto& buf : buffCopies) { buf->bindDeviceMemory(nullptr, 0); }
-	auto imgCopies = _imageMemoryBindings;
-	for (auto& img : imgCopies) { img->bindDeviceMemory(nullptr, 0); }
-
-	[_mtlBuffer release];
-	_mtlBuffer = nil;
+	if (_mtlTexture) {
+		[_mtlTexture release];
+		_mtlTexture = nil;
+
+		// Having no buffer and texture being host accessible means we allocated memory for the mapping
+		if (!_mtlBuffer && mvkIsAnyFlagEnabled(_vkMemPropFlags, VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT)) {
+			free(_map);
+			_map = nullptr;
```

Comment:
Where is `_map` set if there is no `_mtlBuffer`?

And if we're going to set it to `nullptr`, we should move that part below, so it's also set to `nullptr` for the case where we have a `_mtlBuffer`.